File: server/posts/models/posts.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Post struct {
	ID         uint           `json:"id" gorm:"primaryKey"`
	CreatedAt  time.Time      `json:"createdAt"`
	UpdatedAt  time.Time      `json:"updatedAt"`
	DeletedAt  gorm.DeletedAt `json:"-" gorm:"index"`
	UserID     uint           `json:"creatorId"`
	StateValue uint16         `json:"stateValue" gorm:"-:migration"`
	Title      string         `json:"title"`
	Body       string         `json:"body"`
	Likes      uint           `json:"likes" gorm:"default=0"`
	Comments   []Comment      `gorm:"constraint:OnUpdate:CASCADE,OnDelete:CASCADE;foreignKey:PostID;references:ID;"`
	Liked      []Like
}

type PaginatedPosts struct {
	Posts   []Post `json:"posts"`
	HasMore bool   `json:"hasMore"`
}

File: web/src/requests/posts.ts
export interface Post {
  id: number;
  createdAt: string;
  updatedAt: string;
  creatorId: number;
  stateValue: 0 | 1;
  title: string;
  body: string;
  likes: string;
}

export interface PaginatedPosts {
  posts: Post[];
  hasMore: boolean;
}

const url = process.env.API_URL
const headers = {
  Accept: "application/json",
  "Content-Type": "application/json",
}


export const CreatePost = async (
  title: string,
  body: string
): Promise<Post> => {
  const res = await fetch(`${url}/posts`, {
    method: "POST",
    body: JSON.stringify({
      title: title,
      body: body,
    }),
    headers: headers,
    credentials: "include",
  });

  if (!res.ok) {
    const err = await res.json();
    throw new Error(err);
  }

  return await res.json();
};

export const likePost = async (id: number, value: 0 | 1) => {
  const res = await fetch(`${url}/posts/${id}/like`, {
    method: "POST",
    body: JSON.stringify({
      value: value
    }),
    headers: headers,
    credentials: "include",
  });
  
  if (!res.ok) {
    const err = await res.json();
    throw new Error(err);
  }

  return await res.json();
}